### PR TITLE
LFS: make HTTP auth period configurable

### PR DESCRIPTION
**Pull request by InonS**
Authentication period is hard-coded for 5 minutes. For LFS this may be too short, so we made  the HTTP auth period configurable.

### CODE REVIEW

**Review by jonasfranz — APPROVED** [review]

**Comment by jonasfranz** [comment]
Please run `make fmt`.

**Comment by InonS** [comment]
@JonasFranzDEV I pushed the commit which removes the semicolon, running `make fmt` changes nothing, but drone keeps referrencing the previous commit. How can we trigger drone ro rerun on this PR?

**Comment by InonS** [comment]
@JonasFranzDEV how can we re-trigger drone to run now that I committed the semicolon fix?

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -115,6 +115,7 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
 - `LFS_START_SERVER`: **false**: Enables git-lfs support.
 - `LFS_CONTENT_PATH`: **./data/lfs**: Where to store LFS files.
 - `LFS_JWT_SECRET`: **\<empty\>**: LFS authentication secret, change this a unique string.
+- `LFS_HTTP_AUTH_EXPIRY_MINUTES`: **5**: LFS authentication validity period in minutes, pushes taking longer than this may fail.
```
Comment (techknowlogick) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]: Can you also add this to `/custom/conf/app.ini.sample`

**Review by techknowlogick — CHANGES_REQUESTED** [review]

**Comment by lafriks** [comment]
I think default should be increased to at least 20 minutes or half hour as I think this really does create problems for it being so small

File: modules/setting/setting.go [hunk 2]
```diff
@@ -136,10 +136,11 @@ var (
...
 	LFS struct {
-		StartServer     bool   `ini:"LFS_START_SERVER"`
-		ContentPath     string `ini:"LFS_CONTENT_PATH"`
-		JWTSecretBase64 string `ini:"LFS_JWT_SECRET"`
-		JWTSecretBytes  []byte `ini:"-"`
+		StartServer           bool   `ini:"LFS_START_SERVER"`
+		ContentPath           string `ini:"LFS_CONTENT_PATH"`
+		JWTSecretBase64       string `ini:"LFS_JWT_SECRET"`
+		JWTSecretBytes        []byte `ini:"-"`
+		HTTPAuthExpiryMinutes int    `ini:"LFS_HTTP_AUTH_EXPIRY_MINUTES"`
```
Comment (sapk) [inline modules/setting/setting.go]: I find more logic and allow more flexible configuration to set this directly as time.Duration you can use mirror option like MIN_INTERVAL as an example and use a more short option name like LFS_HTTP_AUTH_EXPIRY.

**Review by sapk — CHANGES_REQUESTED** [review]
time.Duration is a better format for expiry time option.

**Review by lafriks — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by lafriks** [comment]
I think we should move this to 1.5 as it is quite safe change and should increase stability of LFS

**Comment by techknowlogick** [comment]
@lafriks Sounds good. Let me do that. I had just moved it in preparation of release, however I agree that this would be good to get in sooner.

**Comment by lafriks** [comment]
@sapk need your approval

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 3]
```diff
@@ -115,6 +115,7 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
 - `LFS_START_SERVER`: **false**: Enables git-lfs support.
 - `LFS_CONTENT_PATH`: **./data/lfs**: Where to store LFS files.
 - `LFS_JWT_SECRET`: **\<empty\>**: LFS authentication secret, change this a unique string.
+- `LFS_HTTP_AUTH_EXPIRY_MINUTES`: **20**: LFS authentication validity period in minutes, pushes taking longer than this may fail.
```
Comment (sapk) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]: The doc isn't updated to new duration format.

**Review by sapk — CHANGES_REQUESTED** [review]
Code is good. The documentation need to be updated to the new duration format. Near to be merged :-).

**Comment by lunny** [comment]
@InonS maybe you could follow @sapk 's advice

**Comment by lunny** [comment]
@sapk need your approval.

**Review by sapk — APPROVED** [review]
Thanks, for this good PR.

**Review by appleboy — APPROVED** [review]

**Comment by ohwgiles** [comment]
Why was this line introduced?
https://github.com/go-gitea/gitea/pull/4035/commits/ebdab8976e22b9b5f751e034a2e209354558e047#diff-60f5fca6d29eafbb30eb645a0cd4041fR832

It separates the new configuration variable from the other `LFS_` variables

**Comment by InonS** [comment]
@ohwgiles points out that the config section should, in fact, still be in accordance with the "server" struct,  which contains the "LFS" struct. I should just remove the line
```
sec = Cfg.Section("LFS")
```

@lafriks @lunny @techknowlogick @appleboy @sapk , would any of you mind doing this directly on `master`? It doesn't sound like something warranting an entirely new PR.

**Comment by sapk** [comment]
@ohwgiles Good catch. Thanks

@InonS It is best to do a PR as it will follow all checks (CI, validation, ...) and keep history of why change has been maid. (And I personally doesn't have push right ^^)

**Pull request merged**